### ISSUE TITLE
Forbid to use `type(...) == '...'`, use `isinstance` instead

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#690]
# Rule request

## Thesis

Comparing types with `type(some) == 'builtin.str'` is incorrect.
We should use `isinstance(some, str)`

## Reasoning

This is a refactoring rule which enforce best practice for python code.
Using `type()` in this way is unpythonic.

**Comment by serhii73** [comment]
![Screenshot from 2019-08-18 12-41-36](https://user-images.githubusercontent.com/24910277/63222850-a13ece80-c1b5-11e9-9123-120532416859.png)
Sorry, https://deepsource.io/blog/python-common-mistakes/ - recommends use `type` =)
What do you think about that @sobolevn?

**Comment by sobolevn** [comment]
There's an important point here: you should not do this at all.

Correct solution:

```python
def which_number_type(num: int) -> None:
    ...
```

And run `mypy` to make sure your types is correct. In this case `isinstance` is more than enough. And using `type()` for compare is invalid.

**Comment by serhii73** [comment]
![Screenshot from 2019-08-18 23-22-30](https://user-images.githubusercontent.com/24910277/63229962-22767f80-c20f-11e9-9e40-79f9988bd573.png)
mypy don't show an error in this case. 
Maybe use `type` make sense?
